src/Libs/GLib-2.0/Public/Dir.cs

```diff
@@ -9,4 +11,9 @@ public static Dir Open(string path, uint flags)
 
         return new Dir(handle);
     }
+
+    public void Dispose()
+    {
+        _handle.Dispose();
```

Comment:
Thanks for a review. I just have one question, why `as IDisposable` in test is needed ? Since Dir already uses IDisposable interface.

Reply:
If the interface would be removed by accident nobody would notice.

I updated the unit test code to avoid a null warning.

Reply:
Okay, that makes sence. I've commited requested changes.

Reply:
I updated my example for the unit test to a direct cast of the `IDisposable` interface, can you implement it like that? This avoids a null reference warning in the unit test as the `as` operator could return `null`.
```CS
var dir = (IDisposable) Dir.Open(".", 0);
```

Did you see my comment above this conversation? I asked to not merge main into this branch but rebase onto main instead. If I would merge your PR now, there would be 3 commits in the history instead of one. Now as you merged the changes, can you please apply the mentioned change and afterwards squash all your commits into a single one so the git history is kept clean?

Reply:
Sorry. I had to hard reset everything, but is it okay now ?

Reply:
The code looks good now 👍.

In the commit message you wrote "Close hangle"? I don't understand it. Can you just remove it? The rest of the message is ok.

You don't need to reset everything for it. See the git "amend" feature: https://www.atlassian.com/git/tutorials/rewriting-history

Reply:
But amed just created new commits in the history that I can't squash because of the newly created merge commit. I'm just gonna reset everything lol.

Reply:
I'm very sorry for wasting your time. I think now everything will be alright.

Reply:
Unfortunately the code is not the same as before.

In the dispose method you wrote `Handle.Dispose()` instead of `_handle.Dispose()`

Let me guide you through the process:

1. In your branch with your code apply the change with the variable name.
2. Stage changes: `git add ./Libs/GLib-2.0/Public/Dir.cs`
3. Amend commit:`git commit --amend --no-edit`
4. Force push changes to github

Reply:
Okay it worked, thanks.

Reply:
Great it worked :+1: 

If you have a Git UI they often support a lot of those features (some more, some less). It is worth to get familiar with git and it's tools as it is a massive time saver to  get a clean history.

Reply:
Again, sorry for wasting a time, thought would be simple commit and that's it, but learned a lot, thanks to you.
I'm using Github Desktop, but I released that I need to use git in CLI, because I know nothing about how git works.

Reply:
I would assume that Github Desktop is not a very bad tool. It supports squashing and amending and [more](https://docs.github.com/en/desktop):

- https://docs.github.com/en/desktop/contributing-and-collaborating-using-github-desktop/managing-commits/squashing-commits
- https://docs.github.com/en/desktop/contributing-and-collaborating-using-github-desktop/managing-commits/amending-a-commit

I don't know if using the commandline makes a huge difference. The important thing is to try to understand what is happening in git. Once understood it is quite easy to manage your code. Then you "only" need to master some kind of tool (either the CLI interface or an application) to make git do what you want.

Reply:
I'm not saying that Github Desktop is a bad tool, I think it's a great tool, but at least for me while using Github Desktop, I did not had an incentive to understand what is happening in git until now. But using git in cli will for sure have direct or indirect push for mastering git.